src/Viewer.tsx

```diff
@@ -423,19 +425,28 @@ function Viewer(): ReactElement {
       }
       isLoadingInitialDataset.current = true;
       let newCollection: Collection;
-      const collectionUrlParam = initialUrlParams.collection;
-      const datasetParam = initialUrlParams.dataset;
       let datasetKey: string;
-
-      if (datasetParam && urlUtils.isUrl(datasetParam) && !collectionUrlParam) {
-        // Dataset is a URL and no collection URL is provided;
-        // Make a dummy collection that will include only this dataset
-        newCollection = Collection.makeCollectionFromSingleDataset(datasetParam);
-        datasetKey = newCollection.getDefaultDatasetKey();
+      // Check if we were passed a collection + dataset from the previous page.
+      if (location.state && "collection" in location.state && "datasetKey" in location.state) {
```

Comment:
Suggestion: this amount of conditionals is something I would usually move to a function that makes it clear what you're testing to make it easier to comprehend. ie `locationHasCollectionAndDataset: () => boolean` or something more succinct :)